src/main/kotlin/com/dprint/services/DprintService.kt

```diff
@@ -245,23 +247,31 @@ class DprintService(private val project: Project) {
     }
 
     fun canFormat(filePath: String): Boolean {
+        var status = 0
         LOGGER.info(Bundle.message("formatting.checking.can.format", filePath))
         getEditorService()?.let { editorService ->
             val stdin = editorService.outputStream
             val stdout = editorService.inputStream
 
-            writeInt(stdin, CHECK_COMMAND)
-            writeString(stdout, stdin, filePath)
-            writeSuccess(stdin)
+            synchronized(this) {
```

Comment:
Will be so much nicer once I implement https://github.com/dprint/dprint/pull/482

I'm going to start on that one soon.